src/plugins/layout.ts

```diff
@@ -198,15 +201,14 @@ export default class Layout implements IPlugin {
     return layoutState;
   }
 
-  private watch(element: Element, layoutState: IElementLayoutState) {
+  private watch(element: Element, styles: CSSStyleDeclaration, layoutState: IElementLayoutState) {
 
     // We only wish to watch elements once and then wait on the events to push changes
     if (this.watchList[layoutState.index]) {
       return;
     }
 
     // Check if scroll is possible, and if so, bind to scroll event
-    let styles = window.getComputedStyle(element);
     let scrollPossible = (layoutState.layout !== null
                           && (styles["overflow-x"] === "auto"
```

Comment:
Possible null dereference here  if styles == null. Didn't check the entire code to see if that was the case.

Reply:
I've reverted these lines to keep things simple. And do we need to evaluate all styles? Unfortunately yes, no other straight forward way to compute only one property.  And already profiling in the new iteration.